site/en/docs/extensions/migrating/rhc/index.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: 'layouts/doc-post.njk'
+title: "Dealing with Remote Hosted Code Violations"
+description: A reference for what to do when your extension is rejected from the Chrome Web Store becaue of Remotely Hosted Code.
+date: 2023-11-23
+updated: 2023-11-23
+---
+
+## What is Remotely Hosted Code?
+
+Remotely Hosted Code, or RHC, is what the Chrome Web Store calls anything that is executed by the browser that is loaded from someplace other than the extension's own files. Things like javascript, wasm, etc. It **_does not_** include data - things like JSON or CSS.
+
+
+## Why is RHC no longer allowed?
+
+With Manifest v3 extensions now need to bundle _all_ code they are using inside the extension itself. In the past, you could dynamically inject script tags from any URL on the web.
+
+
+## I was told my extension has RHC - what's going on?
+
+If your extension was rejected during review with a [Blue Argon][blue-argon] error, then our reviewers believe that your extension is currently using Remotely Hosted Code. This is usually the result of an extension trying to add a script tag  with a remote resource (i.e. from the open web, rather than the files included in the extension), or <code>[fetch][fetch]</code>ing a resource to execute directly.
+
+
+## How to spot RHC
+
+Spotting RHC isn't particularly difficult once you know what to look for. A great first option would be to check for the strings "http://" or "https://" in your project. If you have an RHC violation, then you would likely be able to locate them by finding that. If you have a full build system, or use dependencies from [npm][npm] or other third party sources, make sure you are searching the _compiled_ version of the code, since that is what is being evaluated by the store. If you are still unable to find the problem, then the next step would be to contact [One Stop Support][oss] They will be able to outline the specific violations, and what is needed to get the extension published as soon as possible.
```

Comment:
```suggestion
Spotting RHC isn't particularly difficult once you know what to look for. A great first option would be to check for the strings "http://" or "https://" in your project. If you have an RHC violation, then you would likely be able to locate them by finding that. If you have a full build system, or use dependencies from [npm][npm] or other third party sources, make sure you are searching the _compiled_ version of the code, since that is what is being evaluated by the store. If you are still unable to find the problem, then the next step would be to contact [One Stop Support][oss]. They will be able to outline the specific violations, and what is needed to get the extension published as soon as possible.
```